client/shipping-label/state/reducer.js

```diff
@@ -422,21 +422,24 @@ reducers[ SET_PACKAGE_TYPE ] = ( state, { packageId, boxTypeId } ) => {
 	const newPackages = { ...state.form.packages.selected };
 	const oldPackage = newPackages[ packageId ];
 
+	const oldBox = state.form.packages.all[ oldPackage.box_id ];
+	const newBox = state.form.packages.all[ boxTypeId ];
+	const weight =
```

Comment:
> but it's certainly not expected.

well, and I don't want to sound offensive here, but it _is_ expected because that's what a bunch of people very publicly agreed to and standardized :wink:

it's more that we're often unfamiliar with it because we don't often run into it being a problem. in general fixed-point arithmetic has for most of programming history been a thing all to itself due to the limits of the floating-point representation. it's _acknowledged_ that all floating-point arithmetic is an approximation and that it also breaks mathematical laws.

yes! floating-point addition is not associative and commutative even though arithmetic addition is. some languages have a specific fixed-point data type (SQL does, for example) and often one is required to specify how many places will be used beyond the decimal. in some more esoteric places we end up storing actual decimal data in BCD (binary-coded decimal) which treats each digit as a nibble or half-word of four bits.

since most of our floating-point math _does_ deal with approximations this normally is safe but it always reminds us that it is not exact, rounding errors propagate, and if we _need_ fixed-point math we need to use integer data types.